DuckDuckGo/MainViewController.swift

```diff
@@ -721,4 +728,110 @@ extension MainViewController: Themable {
+            UIKeyCommand(input: "[", modifierFlags: [.shift, .command], action: #selector(keyboardPreviousTab)),
+            UIKeyCommand(input: "\\", modifierFlags: [.shift, .control], action: #selector(keyboardShowAllTabs)),
+            UIKeyCommand(input: "]", modifierFlags: [.command], action: #selector(keyboardBrowserForward)),
+            UIKeyCommand(input: "[", modifierFlags: [.command], action: #selector(keyboardBrowserBack)),
+            UIKeyCommand(input: "f", modifierFlags: [.alternate, .command], action: #selector(keyboardFind)),
+            UIKeyCommand(input: UIKeyCommand.inputRightArrow, modifierFlags: [.command], action: #selector(keyboardBrowserForward)),
+            UIKeyCommand(input: UIKeyCommand.inputLeftArrow, modifierFlags: [.command], action: #selector(keyboardBrowserBack)),
+            UIKeyCommand(input: UIKeyCommand.inputUpArrow, modifierFlags: [], action: #selector(keyboardArrowUp)),
+            UIKeyCommand(input: UIKeyCommand.inputDownArrow, modifierFlags: [], action: #selector(keyboardArrowDown)),
+            UIKeyCommand(input: UIKeyCommand.inputEscape, modifierFlags: [], action: #selector(keyboardEscape))
+        ]
+    }
+    
+    @objc func keyboardFind() {
+        if let controller = homeController {
+            controller.launchNewSearch()
+        } else {
+            omniBar.becomeFirstResponder()
+        }
+    }
+    
+    @objc func keyboardEscape() {
+        
+        if let controller = autocompleteController {
+            controller.keyboardEscape()
+            homeController?.collectionView.omniBarCancelPressed()
+        } else if let controller = homeController {
+            controller.omniBarCancelPressed()
+        } else {
+            omniBar.resignFirstResponder()
+        }
+    }
+    
+    @objc func keyboardArrowDown() {
+        if let controller = autocompleteController {
+            controller.keyboardMoveSelectionDown()
+        } else {
+            currentTab?.webView.becomeFirstResponder()
+        }
+    }
+    
+    @objc func keyboardArrowUp() {
+        if let controller = autocompleteController {
+            controller.keyboardMoveSelectionUp()
+        } else {
+            currentTab?.webView.becomeFirstResponder()
+        }
+    }
+
+    @objc func keyboardNewTab() {
+        if currentTab != nil {
+            newTab()
+        } else {
+            keyboardFind()
+        }
+    }
+    
+    @objc func keyboardCloseTab() {
+        guard let tab = currentTab else { return }
+        closeTab(tab.tabModel)
+        if tabManager.count == 0 {
+            launchNewSearch()
+        }
+    }
+    
+    @objc func keyboardNextTab() {
+        guard let tab = currentTab else { return }
+        guard let index = tabManager.model.indexOf(tab: tab.tabModel) else { return }
+        let targetTabIndex = index + 1 >= tabManager.model.count ? 0 : index + 1
+        select(tabAt: targetTabIndex)
+    }
```

Comment:
Traversing tabs (next/prev) should dismiss focus from omni bar, otherwise we end up on tab that has still omni bar selected (and with the url from the previous tab)